scripts/segmentation/eval_segmentation.py

```diff
@@ -0,0 +1,172 @@
+    parser.add_argument('--pretrained', action="store_true",
+                        help='whether to use pretrained params')
+    parser.add_argument('--ngpus', type=int, default=0)
+    parser.add_argument('--aux', action='store_true', default=False,
+                        help='Auxiliary loss')
+    # synchronized Batch Normalization
+    parser.add_argument('--syncbn', action='store_true', default=False,
+                        help='using Synchronized Cross-GPU BatchNorm')
+    parser.add_argument('--resume', type=str, default=None,
+                        help='put the path to resuming file if needed')
+    # evaluation only
+    parser.add_argument('--eval', action='store_true', default=False,
+                        help='evaluation only')
+
+    args = parser.parse_args()
+    
+    args.ctx = [mx.cpu(0)]
+    args.ctx = [mx.gpu(i) for i in range(args.ngpus)] if args.ngpus > 0 else args.ctx
+
+    args.norm_layer = mx.gluon.contrib.nn.SyncBatchNorm if args.syncbn \
+        else mx.gluon.nn.BatchNorm
+    args.norm_kwargs = {'num_devices': args.ngpus} if args.syncbn else {}
+    return args
+    
+
+def test(args, model):
+    # output folder
+    outdir = 'outdir'
+    if not os.path.exists(outdir):
+        os.makedirs(outdir)
+    # image transform
+    input_transform = transforms.Compose([
+        transforms.ToTensor(),
+        transforms.Normalize([.485, .456, .406], [.229, .224, .225]),
+    ])
+    data_kwargs = {'transform': input_transform, 'base_size': args.base_size,
+                   'crop_size': args.crop_size}
+    # get dataset
+    if args.eval:
+        testset = get_segmentation_dataset(
+            args.dataset, split='val', mode=args.mode, **data_kwargs)
+    else:
+        testset = get_segmentation_dataset(
+            args.dataset, split='test', mode=args.mode, **data_kwargs)
+    size = len(testset)
+    # get dataloader
+    test_data = gluon.data.DataLoader(
+            testset, args.batch_size, last_batch='keep', shuffle=False, num_workers=args.workers)
+
+    print(model)
+    if not args.eval:
+        evaluator = MultiEvalModel(model, testset.num_class, ctx_list=args.ctx)
+    metric = gluoncv.utils.metrics.SegmentationMetric(testset.num_class)
+
+    tbar = tqdm(test_data)
+    metric.reset()
+    tic = time.time()
+    for i, (batch, dsts) in enumerate(tbar):
+        if args.eval:
+            targets = mx.gluon.utils.split_and_load(dsts, ctx_list=args.ctx, even_split=False)
+            data = mx.gluon.utils.split_and_load(batch, ctx_list=args.ctx, batch_axis=0, even_split=False)
+            outputs = None
+            for x in data:
+                output = model.forward(x)[0]
+                outputs = output if outputs is None else nd.concat(outputs, output, axis=0)
+            outputs = [outputs]
+            metric.update(targets, outputs)
+            pixAcc, mIoU = metric.get()
+            tbar.set_description( 'pixAcc: %.4f, mIoU: %.4f' % (pixAcc, mIoU))
+        else:
```

Comment:
what will happen if not eval only

Reply:
In the current script,  `args.eval=False` is only required for `test` mode, which will convert the input RGB images into segmented images as outputs. However, there are still incompatible issues for INT8 models and I'll try to fix them.